site/public/js/calendar.js

```diff
@@ -97,6 +97,31 @@ function dateToStr(year, month, day) {
     return `${year.toString()}-${month.toString().padStart(2, '0')}-${day.toString().padStart(2, '0')}`;
 }
 
+/**
+ * This function returns a slightly darker color than the color variable name passed.
+ *
+ * @param colorstr : string the color to darken in the form "var(--color-name)"
+ * @returns {string} a hex code for a slightly darker shade
+ */
+function darken(colorstr) {
```

Comment:
What happens if the color passed into this function is like the darkest black. What will be returned by this function? Will it still work?

And if not how can we try to fix this? Also, when I switch to dark mode, I feel like the change in color is just not noticeable enough, but that might just be me.

Reply:
I see what you're saying about the color change being very subtle on dark mode. I just doubled the amount the function darkens colors by. Let me know if that's better.

The function should still work for black and very dark grays. It doesn't alter values that are '0' or '1', so it would just leave black as-is. I can make it return a slightly lighter gray, but there aren't any calendar buttons that use black or very dark gray backgrounds so I don't know if it's necessary.

Reply:
It could be good to have the function account for a very dark color even if we don't have to deal with those scenarios for now, just in case we run into that situation on future PRs.